### PR TITLE
Feature/refactor user service as observable store

### PR DESCRIPTION
**Pull request by dtslvr**

### CODE REVIEW

**Comment by dtslvr** [comment]
Hey @cnschwarz

I'm using `@codewithdan/observable-store` for the first time. Would be happy to get a feedback from you! 😃

File: apps/client/src/app/app.component.ts [hunk 1]
```diff
@@ -64,26 +65,22 @@ export class AppComponent implements OnDestroy, OnInit {
...
+          this.canCreateAccount = hasPermission(
+            this.user.permissions,
+            permissions.createUserAccount
+          );
+        } else if (!this.tokenStorageService.getToken()) {
+          // User has not been logged in
           this.user = null;
         }
+
+        this.cd.markForCheck();
```
Comment (vzickner) [inline apps/client/src/app/app.component.ts:83]: Rename `cd` to `changeDetector`

**Review by vzickner — APPROVED** [review]

File: apps/client/src/app/app.component.ts (hunk 1 above)
Comment (dtslvr) [reply apps/client/src/app/app.component.ts:83]: Thanks, I have refactored it here: #118

**Comment by cnschwarz** [comment]
Hey @dtslvr 

I had a look, looks good to me. 👍🏽  
This is how I used it also for certain cases in my app. Much more lightweight than the usage of ngrx. Especially with the current level of complexity of the app.

**Comment by dtslvr** [comment]
> Hey @dtslvr
> 
> I had a look, looks good to me. 👍🏽
> This is how I used it also for certain cases in my app. Much more lightweight than the usage of ngrx. Especially with the current level of complexity of the app.

Thanks for your feedback @cnschwarz. Yes, that's also my first impression. The use of `@codewithdan/observable-store` is straightforward and predictable.

**Pull request merged**